pdns/dolog.hh

```diff
@@ -88,8 +92,22 @@ void genlog(int level, const char* s, Args... args)
 {
   std::ostringstream str;
   dolog(str, s, args...);
+
   if(g_syslog)
     syslog(level, "%s", str.str().c_str());
+
+#ifdef DNSDIST
+  if (g_logtimestamps) {
+    char buffer[50] = "";
+    struct tm tm;
+    time_t t;
+    time(&t);
+    localtime_r(&t, &tm);
+    strftime(buffer, sizeof(buffer), "%b %d %H:%M:%S ", &tm);
```

Comment:
We might want to check the return value of `strftime` before reading `buffer` since POSIX states that: `If the total number of resulting bytes including the terminating null byte is not more than maxsize, [...]. Otherwise, 0 is returned and the contents of the array are indeterminate`.

Reply:
Good idea! I added a commit fixing that for auth, rec and dnsdist logging (the code above is copied from auth/rec). We have a few other unchecked `strftime` uses but I think that's for a separate PR.